tools/webtransport/h3/webtransport_h3_server.py

```diff
@@ -118,11 +118,13 @@ def _h3_event_received(self, event: H3Event) -> None:
 
             method = headers.get(b":method")
             protocol = headers.get(b":protocol")
-            if method == b"CONNECT" and protocol == b"webtransport":
+            origin = headers.get(b"origin")
+            # Accept any Origin but the client must send it.
+            if method == b"CONNECT" and protocol == b"webtransport" and origin:
                 self._session_stream_id = event.stream_id
                 self._handshake_webtransport(event, headers)
             else:
-                self._send_error_response(event.stream_id, 400)
+                self._send_error_response(event.stream_id, 404)
```

Comment:
Updated per https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-04.html#section-3.3-3

@vasilvv Which status code a server should use when the server rejects a request due to origin checks?

Reply:
I believe 403 is the most appropriate error code here.

Reply:
Updated to use 403 for origin check failure. Thanks!